cmake/FindMatlab.cmake

```diff
@@ -1,4 +1,5 @@
-# Vendored from https://gitlab.kitware.com/cmake/cmake/-/raw/v3.17.0/Modules/FindMatlab.cmake
+# Vendored from https://gitlab.kitware.com/cmake/cmake/-/raw/v3.20.0-rc3/Modules/FindMatlab.cmake
+# With local modification to implement https://gitlab.kitware.com/cmake/cmake/-/issues/21912
```

Comment:
We need to document this change in Changelog and also the README so that we remember that we are not using a vanilla FindMatlab.cmake. Also the Matlab versions that we support and  this is affecting .

Reply:
> Also the Matlab versions that we support and this is affecting .

This is something that we could do, but to be honest I don't know which MATLAB version we explicitly support (as we don't test them) so I don't know how can we do this in a proper way.

Reply:
I was actually thinking was about the developers. iDynTree is a project that a few developers we know use as a reference for software engineering. In case some other project faces something similar and come looking for answers here. That was the scenario I had in mind.